osu.Game.Rulesets.Osu/Mods/OsuModBlindTravel.cs

```diff
@@ -0,0 +1,227 @@
+        };
+
+        private Double cameraDelay => CameraDelay.Value;
+
+        private float applyIncreasedVisibilityDuration = 1000;
+
+        private float increasedVisibilityDuration = 1000;
+
+        private const float default_scope_scale = 1;
+
+        private const float default_scope_fov = 0.6f;
+
+        [SettingSource("Base FOV", "Adjust the base field of view")]
+        public BindableFloat BaseScopeFov { get; } = new BindableFloat(default_scope_fov)
+        {
+            MinValue = default_scope_fov,
+            MaxValue = default_scope_scale,
+            Precision = 0.05f
+        };
+
+        private float baseScopeFOV => BaseScopeFov.Value;
+
+        private float currentScopeFOV = default_scope_fov;
+
+        // scaling works as (if fov = 1; scale is half of that)
+        private float CurrentScopeFovAsScale => 1 / currentScopeFOV;
+
+        [SettingSource("Change FOV based on combo", "Shrinks the FOV based on combo")]
+        public BindableBool ComboBasedFOV { get; } = new BindableBool(true);
+
+        private const float shrinkFOVWithComboBy = 0.05f;
+
+        private Playfield? playfield;
+
+        private ParallaxContainer? parallaxContainer;
+
+        private float baseParallaxAmount;
+
+        public void ApplyToPlayer(Player player)
+        {
+            baseParallaxAmount = player.BackgroundParallaxAmount;
+            parallaxContainer = player.FindClosestParent<OsuScreenStack>().parallaxContainer;
+
+            currentScopeFOV = baseScopeFOV;
+            applyParallaxForFOV(parallaxContainer, currentScopeFOV);
+
+            player.IsBreakTime.ValueChanged += onBreakTime;
+        }
+
+        private void onBreakTime(ValueChangedEvent<bool> e)
+        {
+            InternalApplyIncreasedVisibilityState();
+        }
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
+        {
+            // we avoid assiging to playfield every update by assigning it here.
+            playfield = drawableRuleset.Playfield;
+        }
+
+        public void Update(Playfield playfield)
+        {
+            Debug.Assert(parallaxContainer != null);
+
+            scheduler.Update();
+
+            var cursorPos = playfield.Cursor.ActiveCursor.DrawPosition;
+
+            // applies parallax to managed cursors (such as auto).
+            parallaxContainer.MousePosition = cursorPos;
+
+            if (!isTraveling)
+                return;
+
+            var scopePosition = -cursorPos;
+
+            moveDrawableToScope(playfield, scopePosition, playfield);
+        }
+
+        private void moveDrawableToScope(Drawable drawable, Vector2 position, Playfield playfield)
+        {
+            var scopePositionForDrawable = position + playfield.DrawSize / 2;
+
+            drawable.MoveTo(scopePositionForDrawable, cameraDelay);
+        }
+
+        private bool increasedVisibilityMode;
+
+        private Scheduler scheduler = new Scheduler();
```

Comment:
This should not be required.

Reply:
I think i get all the other request, but how would i not need a scheduler for that? isn't it the simplest way

Reply:
Sorry for this double mention, messed up not switching accounts.
But how would the scheduler not be required like, it seems to be the simplest way to schedule something for the future isn't it?

Reply:
We don't really add a scheduler anywhere just for this kind of purpose.

You can just store a timestamp in this case, should work quite well.